include/alpaka/mem/buf/BufUniformCudaHipRt.hpp

```diff
@@ -468,6 +468,18 @@ namespace alpaka
                 meta::DependentFalseType<TElem>::value,
                 "HIP/CUDA devices support only one-dimensional stream-ordered memory buffers.");
         };
+#    if defined(ALPAKA_ACC_GPU_CUDA_ENABLED) && (CUDA_VERSION >= 11020)
+        //! The CUDA/HIP stream-ordered scalar memory allocation capability trait specialization.
+        template<>
+        struct IsAsyncBufferSupported<DimInt<0u>, DevUniformCudaHipRt> : public std::true_type
+        {
+        };
+        //! The CUDA/HIP stream-ordered 1D memory allocation capability trait specialization.
+        template<>
+        struct IsAsyncBufferSupported<DimInt<1u>, DevUniformCudaHipRt> : public std::true_type
```

Comment:
Currently no, because the implementation is based on `cudaMallocAsync`, and there are no 2D/3D counterparts.
If there is interest, the 2D/3D/ND versions could be implemented "by hand" on top of the linear memory returned by `cudaMallocAsync` and making some conservative assumptions about what `pitch to use, kind of like it's done in the CPU case.

Reply:
No, the default behaviour should be fine for now. 1D it is then.